test/generateTasks.spec.js

```diff
@@ -239,14 +238,14 @@ describe('generateTasks', () => {
     expect(prjTask).toEqual({
       pattern: '*.js',
       commands: 'my-cmd',
-      fileList: [`${workDir}/prj/test.js`].map(path.normalize)
+      fileList: [`test.js`].map(path.normalize)
     })
 
     const parentFolderTask = result.find(item => item.pattern === '../outside/*.js')
     expect(parentFolderTask).toEqual({
       pattern: '../outside/*.js',
       commands: 'my-cmd',
-      fileList: [`${workDir}/outside/test.js`, `${workDir}/outside/test2.js`].map(path.normalize)
+      fileList: [`../outside/test.js`, `../outside/test2.js`].map(path.normalize)
```

Comment:
I found why absolute paths were introduced. Here is the commit: https://github.com/bnjjj/lint-staged/commit/41ae0cb3e36674a94afd69d15cafd43eb69e5969

Can you please ensure we don’t break this use case? We should add a test if it doesn’t exist.

Reply:
I'm sorry for the confusion. I believe the case we're talking about is then the git root and `package.json` location aren't the same. Imagine following scenario:

```
.git/
subdir/
|-package.json
|-node_modules
``` 
in this case, linters will be installed into `subir/node_modules` and passing paths relative to `.git/` to them is not going to work, since they are going to be executed relative to `subdir/`. That's why _I believe_ we have introduced absolute paths in the first place. The more I think about it, the more I tend to keep it like this since I don't know how to remove the ambiguity with this scenario. 

Do you have any ideas? If not, I'm sorry for the extra work but I think then we need to revert to 
d770672 and add it as an option. Thoughts?

Reply:
thanks to this line https://github.com/okonet/lint-staged/blob/master/src/generateTasks.js#L33 and yes it works for my case because my linter is located in `ui/node_modules` and all my files need to be linted are in `ui/`